src/main/java/org/opensearch/ad/MemoryTracker.java

```diff
@@ -248,21 +254,38 @@ public long estimateTotalModelSize(AnomalyDetector detector, int numberOfTrees,
      *  56 + # trees * (2248 + 152 + 6120 + 104 + (1040 + 255* (dimension * 4 * 2 + 64)) * adjusted bounding box cache ratio) +
      *  (256 * # trees  * 2 + 256 * # trees * dimension) * 4 bytes  * 0.5 + 1064 + 24 + 24 + 16
      *  = 56 + # trees * (8624 + (1040 + 255 * (dimension * 8 + 64)) * adjusted bounding box cache ratio) + 256 * # trees *
-     *   (2 + dimension) * 4 * 0.5 + 1128
+     *   (3 + dimension) * 4 * 0.5 + 1128
      *
      * @param dimension The number of feature dimensions in RCF
      * @param numberOfTrees The number of trees in RCF
      * @param boundingBoxCacheFraction Bounding box cache ratio in RCF
      * @return estimated RCF model size
      */
     public long estimateRCFModelSize(int dimension, int numberOfTrees, double boundingBoxCacheFraction) {
-        float averagePointStoreUsage = dimension == 1 ? 1 : 0.5f;
-        float actualBoundingBoxUsage = boundingBoxCacheFraction >= 0.3 ? 1 : (float) boundingBoxCacheFraction;
+        double averagePointStoreUsage = 0;
+        int logNumber = dimension + 1;
+        if (dimension >= 32) {
+            averagePointStoreUsage = 1.0d / (log2(logNumber) + Math.log10(logNumber));
+        } else {
+            averagePointStoreUsage = 1.0d / log2(logNumber);
+        }
+        double actualBoundingBoxUsage = boundingBoxCacheFraction >= 0.3 ? 1d : boundingBoxCacheFraction;
         long compactRcfSize = (long) (56 + numberOfTrees * (8624 + (1040 + 255 * (dimension * 8 + 64)) * actualBoundingBoxUsage) + 256
-            * numberOfTrees * (2 + dimension) * 4 * averagePointStoreUsage + 1128);
+            * numberOfTrees * (3 + dimension) * 4 * averagePointStoreUsage + 1128);
         return compactRcfSize;
     }
 
+    /**
+     * Function to calculate the log base 2 of an integer
+     *
+     * @param N input number
+     * @return the base 2 logarithm of an integer value
+     */
+    public static double log2(int N) {
```

Comment:
minor. this function log2 is not part of the MemoryTracker interface. it can be moved to some other util class or be private.

Reply:
moved to MathUtil.